overlord/auth/auth.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package auth
+
+import (
+	"fmt"
+
+	"github.com/ubuntu-core/snappy/overlord/state"
+)
+
+// AuthState represents current authenticated users as tracked in state
+type AuthState struct {
+	LastID int        `json:"lastId"`
+	Users  []AuthUser `json:"users"`
+}
+
+// AuthUser represents an authenticated user
+type AuthUser struct {
+	ID         int      `json:"id"`
+	Username   string   `json:"username,omitempty"`
+	Macaroon   string   `json:"macaroon,omitempty"`
+	Discharges []string `json:"discharges,omitempty"`
+}
+
+// NewUser tracks a new authenticated user and saves its details in the state
+func NewUser(st *state.State, username, macaroon string, discharges []string) (*AuthUser, error) {
+	authenticatedUser := AuthUser{
+		ID:         1,
```

Comment:
We can handle the internal side of the data properly even if in practice we end up using a single user. We just need to load the existing AuthState here if it exists, and only initialize it if it doesn't. In both cases, the ID is LastId+1.